BUILD.md

```diff
@@ -58,7 +58,7 @@ $ PARALLEL_LEVEL=8 ./build.sh libcuml  # build and install libcuml limiting para
 $ ./build.sh libcuml -n                # build libcuml but do not install
 $ ./build.sh prims --allgpuarch        # build the ML prims tests for all supported GPU architectures
 $ ./build.sh cuml --singlegpu          # build the cuML python package without MNMG algorithms
-$ ./build.sh --ccache                  # use ccache to cache compilations, speeding up subsequent builds
+$ ./build.sh --sccache                  # use sccache to cache compilations, speeding up subsequent builds
```

Comment:
Can we keep both options? `sccache` only really benefits docker users due to the absolute path requirement, `ccache` can still be benefitial for some developers

Reply:
I agree we should keep both options by transforming the flag into an option.

```
./build.sh --cache-tool=[ccache|sccache]
```

Reply:
Since `sccache` defaults to a local cache in the absence of a remote configuration, would that be sufficient? Or do you guys still think we should include both `sccache` and `ccache`?

Reply:
We're going to be removing `ccache` from our CI containers in favor of `sccache`. So if users are going to use our containers to build, then `sccache` will already be configured and they'll benefit from the shared cache by using absolute paths in the containers. If they build outside of our containers, then wouldn't `sccache` just behave similarly to `ccache` anyway?

Reply:
> If they build outside of our containers, then wouldn't sccache just behave similarly to ccache anyway (i.e. use a local cache)?

We shouldn't presume that sccache is sufficient to all developers / people building from source. sccache has more restrictions ( narrower compiler support, abs paths ) and these could make it a non-starter for these people. By making the option flexible by having the user specify the name instead of hardcoding it we allow more use cases at no real cost.

Reply:
@dantegd @robertmaynard Is it cool if we revert these changes made to `BUILD.md` and the associated files, and add them as a future enhancement? This way we don't hold up the rest of the PR from being merged.

Reply:
Sounds good by me, @AyodeAwe can you open a GH issue to track?

Reply:
Yes @dantegd. See [here](https://github.com/rapidsai/ops/issues/2041)

I will push the reverts now and await your further review/approval.